include/deal.II/lac/petsc_parallel_vector.h

```diff
@@ -223,11 +223,16 @@ namespace PETScWrappers
                        const size_type     local_size);
 
       /**
-       * Construct a new parallel ghosted PETSc vector from an IndexSet.
-       * Note that @p local must be ascending and 1:1.
-       * The global size of the vector is determined by local.size().
-       * The global indices in @p ghost are supplied as ghost indices
-       * that can also be read locally.
+       * Construct a new parallel ghosted PETSc vector from IndexSets.
+       *
+       * Note that @p local must be ascending and 1:1, see
+       * IndexSet::is_ascending_and_one_to_one().  In particular, the DoFs in
+       * @p local need to be contiguous, meaning you can only create vectors
+       * from a DoFHandler with several finite element components if they are
+       * not reordered by component (use a PETScWrappers::BlockVector
+       * otherwise).  The global size of the vector is determined by
+       * local.size().  The global indices in @p ghost are supplied as ghost
+       * indices that can also be read locally.
```

Comment:
-> so that *they* can ...

Reply:
Because the vector is ghosted, does that mean that it's read-only? If so, maybe we can say that? (Assuming that `ghost` has elements that are not in `local`.)

Reply:
It is actually always a ghosted vector and read-only even if that set is empty.